src/react/extendApp.js

```diff
@@ -12,4 +12,15 @@ export default function extendApp(App) {
       </Provider>
     );
   };
+
+  ['beforeMount', 'afterMount', 'beforeUnmount'].forEach((hookName) => {
+    App.prototype[hookName] = function lifecycleHook(...args) {
```

Comment:
just to be clear, here are we doing this? brain had a hard time parsing the code 😅 

* extract hook by name
* if not a function, default to noop
* run the function against `app`'s context with the given arguments, and return its output
* if successfully run, set the function as a method in the `app` instance